lib/pure/asyncdispatch.nim

```diff
@@ -828,9 +882,9 @@ when defined(windows) or defined(nimdoc):
 
     var dataBuf: TWSABuf
     dataBuf.buf = data
-    dataBuf.len = data.len.ULONG
     GC_ref(data) # we need to protect data until send operation is completed
                  # or failed.
+    dataBuf.len = data.len.ULONG
```

Comment:
Because it was in the first version of this PR and after merging PR 4680, I had to resolve the conflicts!